test/framework/src/main/java/org/elasticsearch/action/support/ActionTestUtils.java

```diff
@@ -77,6 +80,27 @@ public static <T> ActionListener<T> assertNoFailureListener(CheckedConsumer<T, E
         return ActionListener.wrap(consumer, ESTestCase::fail);
     }
 
+    public static <T> ActionListener<T> assertNoSuccessListener(Consumer<Exception> consumer) {
```

Comment:
It would be nice to convert name to remove negation. What about `assertFailureListener` and `assertResultListener`/`assertResponseListener`?

Reply:
I know what you mean, but we're not actually asserting that these things do succeed/fail, just that they _don't_ fail/succeed respectively. We'd need to add in https://github.com/elastic/elasticsearch/issues/108123.